### DISCUSSION TITLE
How to overrride the variables for a model in the variables browser?

### DISCUSSION THREAD

**Discussion by ap766** [OpenModelica/OpenModelica#11377] [Q&A]
So basically I have simulated the model and created the .exe. Now I want to be able to enter the parameters from the command line.
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/54245a6b-431d-4a7c-98df-490db71fec02)

The problem is as you can see in the above image , I have A for tank2 to be 8.0 and V to be 9.0 , however when I open the openmodelica software select the .mat file(ctrl+shift+o) generated due to executing that command in the command line .It doesn't show me 8.0 and 9.0 instead it is 1.0 and 10.0 respectively as you can see in the screenshot below
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/261c6009-69d9-496b-a588-ba4611a33018)

So how do I make 8.0 and 9.0 appear in the variables browser on the right?

**Comment by adeas31** [comment]
It is not possible. They are the override that you can pass to the simulation executable at the runtime. However, you can do the same what you did on the command line. Just override them by changing the values in the `Variables Browser` and then click on re-simulate in the toolbar.

**Reply by ap766** [reply]
Is there any other way possible? Actually I have a GUI where I want to put in the values of the parameters nd I need those values to be as per what has been entered nd not make them enter again

**Reply by adeas31** [reply]
Set them directly in the model then.

**Reply by ap766** [reply]
okay thank you .So directly in the variables browser is the only way. So instead any way to initialise the values in the element parameters(screenshot below)  by using any command similar to what I was using or any way other than direct ?
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/13ca54fc-d685-4cc0-a6e5-4cef6a406b0f)

**Reply by adeas31** [reply]
I am not sure what you want to achieve.

There are commands for modifying parameters. But of course you need to run them every time before simulating the model.

My suggestion is that since you always want to use these values so just hard code them in the model directly and you don't have to override them.

**Reply by arun3688** [reply]
@ap766 Based on what you wrote and some understanding i see that you want to post process the result data to some other external tool, so i suggest you to use OMPython https://openmodelica.org/doc/OpenModelicaUsersGuide/1.21/ompython.html#enhanced-ompython-features or OMJulia depending on your wish , then you can override these parameter and get those values and send it to other tools for post processing and example would 

```
>> from OMPython import ModelicaSystem
>> mod =ModelicaSystem("a.mo", "a")
>> mod.getParameters()
>> mod.setParameters()
>> mod.simulate()
>>mod.getSolutions()
```
This is an example please look into the userguide on how to use the API.

**Reply by casella** [reply]
https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/ompython.html

**Comment by ap766** [comment]
@adeas31 Actually there's a specific requirement.I need to make a PyQt6 Gui that makes input of the parameters from there.And accordingly generate the .mat file such that when the .mat file is opened in openmodelica, the variables browser is overwritten by the values input by the user in the GUI. Could you share what command do you mean ? Earlier i was using TwoConnectedTanks.exe -override=startTime=0,stopTime=5 .But I suppose there isn't any direct command using .exe for the variables browser.So I think i would have to use the model.mo for this purpose and use ompython. But please do let me know what commands do you mean here?

**Reply by adeas31** [reply]
The variables browser shows the values from the `<model>_init.xml` file. So if you want to show the updated values in variable browser then you should also update the xml file.

When you run the executable with `-override` the updated values are stored in the mat file. So if you plot variables you will see the updated values.

Note that if you load a mat file in OMEdit and there is no `<model>_init.xml` file at the same location then you don't see the values textboxes in variables browser.

**Reply by ap766** [reply]
ok Thank you so much! Actually i have the XML file in the same directory 
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/7817cfd0-5404-4fc8-b838-d180b6b630dd)

I notice when I am using overide 
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/52a420b1-5cef-47d3-a5a9-2c144df1f283)

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/502e3e09-d593-458f-baac-c908677d7ae6)
changes are not made to the XML file as you can see.

**Reply by adeas31** [reply]
It is not updated automatically. You need to update it.

**Reply by ap766** [reply]
oh okay.
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/740c26cd-b585-4fd7-ada3-91b9426f6e05)
Basically the documentation here mentions XML file so I thought so.
https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html#

**Reply by ap766** [reply]
okay so i got that init_xml cannot be changed , although the value reflects in the plot(. the simulation will use the new (start, parameter,...) values instead.) , it doesn't reflect in the value column of the variables browser as we haven't changed init_xml

**Comment by ap766** [comment]
Also shouldnt it just work if i am able to modify the XML file?
